src/helper/backend/PamHandle.cpp

```diff
@@ -17,12 +17,38 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  *
  */
+
+#include <cerrno>
+#include <ctime>
+
 #include "PamHandle.h"
 #include "PamBackend.h"
 
 #include <QtCore/QDebug>
 
 namespace SDDM {
+    static void fail_delay(int retval, uint usec_delay, void* appdata_ptr) {
+        constexpr uint uSecMultiplier {1000000}, uSecToNanoSecMultiplier {1000};
```

Comment:
Stylistic nitpick: (feel free to ignore)
I'd just remove these constants and use them directly in the computations, it's not too helpful. Or declare them right before the time computations, so the reader doesn't need to look up their definitions when reading the delay code.